onnxruntime/core/optimizer/reshape_fusion.cc

```diff
@@ -83,42 +78,42 @@ static bool DistilBert_Check(Graph& graph, const Node& concat, const Node& layer
 
   std::vector<graph_utils::EdgeEndToMatch> linear_path{
     {0, 0, "Add", {7}, kOnnxDomain},
-    {0, 0, "MatMul", {1, 9}, kOnnxDomain},
-    {0, 0, "Add", {7}, kOnnxDomain}};
+    {0, 0, "MatMul", {1, 9}, kOnnxDomain}};
   if (!graph_utils::FindPath(reshape, true, linear_path, edges, logger)) {
     return false;
   }
-  const Node& linear_path_root_add = edges[2]->GetNode();
-  if (linear_path_root_add.Index() != layernorm_add.Index()) {
-    return false;
-  }
+
   const Node& linear_path_add = edges[0]->GetNode();
   const Node& linear_path_matmul = edges[1]->GetNode();
 
-  if (layernorm_add.InputDefs().size() < 2) {
+  const Node& node_before_matmul = *graph_utils::GetInputNode(linear_path_matmul, 0);
```

Comment:
GetInputNode might return NULL. Need add a check. Another way is to check NodeArg is same.